src/core/nodeOps.ts

```diff
@@ -180,15 +182,41 @@ export const nodeOps: RendererOptions<TresObject, TresObject | null> = {
       disposeMaterialsAndGeometries(node)
       deregisterCameraIfRequired(node as Object3D)
       deregisterAtPointerEventHandlerIfRequired?.(node as TresObject)
+      invalidateInstance(node as TresObject)
+      node.dispose?.()
+      
     }
 
-    invalidateInstance(node as TresObject)
-    node.dispose?.()
   },
-  patchProp(node, prop, _prevValue, nextValue) {
+  patchProp(node, prop, prevValue, nextValue) {
     if (node) {
       let root = node
       let key = prop
+      if (key === 'object' && prevValue !== null) {
```

Comment:
Is there ever another prop called `object` that will be passed here? Does this need to be more specific, e.g., check if the node is `primitive`?

Reply:
Good catch

Reply:
Done